src/main/java/com/ldtteam/blockui/controls/AbstractTextElement.java

```diff
@@ -492,10 +492,9 @@ public MutableComponent getText()
         return isTextEmpty() ? null : text.get(0);
     }
 
-    @SuppressWarnings({"unchecked", "rawtypes"})
     public void setTextOld(final List<Component> text)
     {
-        setText((List<MutableComponent>)((List) text));
+        setText(text.stream().map(c -> c instanceof MutableComponent m ? m : c.copy()).toList());
```

Comment:
I'm not sure if the instanceof is strictly necessary. It's a minor optimization, I would be fine with calling copy by default.

So
```java
setText(text.stream().map(Component::copy).toList());
```

Same for below

Reply:
This is fine, I will clean up in next porting window if necessary